syntax = "proto3";
import "google/protobuf/empty.proto";
import "common.proto";
option go_package = ".;proto";

service User{
  rpc GetUserPageList(UserFilterPageInfo) returns (UserPageResponse); //用户列表
  rpc GetUserByMobile(MobileRequest) returns (UserResponse); // 通过mobile查询用户
  rpc GetUserById(common.IdRequest) returns (UserResponse); // 通过Id查询用户
  rpc CreateUser(CreateUserRequest) returns (UserResponse); // 创建用户
  rpc DeleteUserByIds (common.IdsRequest) returns (google.protobuf.Empty); // 删除地址
  rpc UpdateUser(UpdateUserRequest) returns (google.protobuf.Empty); // 更新用户信息
  rpc CheckPassword(PasswordCheckRequest) returns (CheckPasswordResponse); // 校验密码
}

message PasswordCheckRequest{
  string password = 1;
  string encryptedPassword = 2;
}

message CheckPasswordResponse{
  bool  success = 1;
}

message MobileRequest{
  string mobile = 1;
}


message CreateUserRequest{
  string nickname = 1;
  string password = 2;
  string mobile = 3;

}

message UpdateUserRequest{
  uint64 id = 1;
  string nickname = 2;
  string gender = 3;
  uint64 birthday = 4;

}

message UserFilterPageInfo{
  common.PageRequest pageRequest = 1; // 分页参数
}
message UserPageResponse{
  common.PageResponse page = 1;
  repeated UserResponse list = 2;
}

message UserResponse{
  uint64 id = 1;
  string nickname = 2;
  string password = 3;
  string mobile = 4;
  uint64 birthday = 5;
  string gender = 6;
  string role = 7;
}